# Exploración de datos: 
## Centros educativos autorizados por el Mineduc y carreras que pueden impartir.

### Librerías

In [1]:
# librerías
import pandas as pd
import numpy as np

### Cargar

In [2]:
df = pd.read_csv("establecimientos.csv", encoding="utf-8-sig")
df.head()

,CODIGO,DISTRITO,DEPARTAMENTO,MUNICIPIO,ESTABLECIMIENTO,DIRECCION,TELEFONO,SUPERVISOR,DIRECTOR,NIVEL,SECTOR,AREA,STATUS,MODALIDAD,JORNADA,PLAN,DEPARTAMENTAL
0,01-01-0003-45,01-110,GUATEMALA,GUATEMALA,LICEO CIENTIFICO CUMORAH,LOTE 572 MANZANA 5 COLONIA MAYA ZONA 18,22604386,ZUHILEM YESENIA ESTRADA ORTIZ,BERLY MARISOL MORALES OCHOA,BASICO,PRIVADO,URBANA,CERRADA TEMPORALMENTE,MONOLINGUE,MATUTINA,DIARIO(REGULAR),GUATEMALA NORTE
1,01-01-0006-45,01-504,GUATEMALA,GUATEMALA,COLEGIO HAN AL AMERICANO,9A. CALLE 28-74 ZONA 14,23661600,DAVID SOTOJ SANCHEZ,OLGA LETICIA PEREZ GARCIA,BASICO,PRIVADO,URBANA,CERRADA DEFINITIVAMENTE,MONOLINGUE,MATUTINA,DIARIO(REGULAR),GUATEMALA NORTE
2,01-01-0010-45,01-620,GUATEMALA,GUATEMALA,COLEGIO LA EDUCACION,7A. AVENIDA 3-20 ZONA 19 COLONIA LA FLORIDA,24310333,RONALD MAURICIO MORALES VIVAR,NATALY GABRIELA DE LEON URÍZAR,BASICO,PRIVADO,URBANA,CERRADA DEFINITIVAMENTE,MONOLINGUE,VESPERTINA,DIARIO(REGULAR),GUATEMALA NORTE
3,01-01-0012-45,01-503,GUATEMALA,GUATEMALA,COLEGIO LA ACADEMIA,13 CALLE 6-91 ZONA 10,23323279,NANCY CAROLINA NAVAS GONZÁLEZ,ANA LUCRECIA LOPEZ FLORES,BASICO,PRIVADO,URBANA,CERRADA DEFINITIVAMENTE,MONOLINGUE,MATUTINA,DIARIO(REGULAR),GUATEMALA NORTE
4,01-01-0014-45,01-403,GUATEMALA,GUATEMALA,"CENTRO EDUCATIVO ""RENUEVO""",6A. 2-34 ZONA 1,55856196,CARLOS HUMBERTO GONZÁLEZ DE LEÓN,JULIO ESTUARDO PÉREZ EUFRAGIO,BASICO,PRIVADO,RURAL,CERRADA DEFINITIVAMENTE,MONOLINGUE,INTERMEDIA,FIN DE SEMANA,GUATEMALA NORTE


In [3]:
df.loc[1000]

CODIGO                                         01-08-8558-45
DISTRITO                                              01-406
DEPARTAMENTO                                       GUATEMALA
MUNICIPIO                                              MIXCO
ESTABLECIMIENTO                COLEGIO 'JESUS REY DE GLORIA'
DIRECCION          36 AV. 4-18 BOSQUES DE SAN NICOLAS ZONA 4
TELEFONO                                     2863760-8027980
SUPERVISOR                  MIGUEL ALVARO RODRIGUEZ PORTILLO
DIRECTOR                                                 NaN
NIVEL                                                 BASICO
SECTOR                                               PRIVADO
AREA                                                  URBANA
STATUS                                 CERRADA TEMPORALMENTE
MODALIDAD                                         MONOLINGUE
JORNADA                                                DOBLE
PLAN                                           FIN DE SEMANA
DEPARTAMENTAL           

In [4]:
pd.DataFrame(df, columns=['DEPARTAMENTO', 'STATUS', 'NIVEL']).head() #es case sensitive

,DEPARTAMENTO,STATUS,NIVEL
0,GUATEMALA,CERRADA TEMPORALMENTE,BASICO
1,GUATEMALA,CERRADA DEFINITIVAMENTE,BASICO
2,GUATEMALA,CERRADA DEFINITIVAMENTE,BASICO
3,GUATEMALA,CERRADA DEFINITIVAMENTE,BASICO
4,GUATEMALA,CERRADA DEFINITIVAMENTE,BASICO


## 3. Análisis del estado inicial de los datos
#### a. Número de registros y variables. 



In [5]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 23302 entries, 0 to 23301
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   CODIGO           23258 non-null  str  
 1   DISTRITO         22210 non-null  str  
 2   DEPARTAMENTO     23258 non-null  str  
 3   MUNICIPIO        23258 non-null  str  
 4   ESTABLECIMIENTO  23253 non-null  str  
 5   DIRECCION        22988 non-null  str  
 6   TELEFONO         20937 non-null  str  
 7   SUPERVISOR       22206 non-null  str  
 8   DIRECTOR         19915 non-null  str  
 9   NIVEL            23258 non-null  str  
 10  SECTOR           23258 non-null  str  
 11  AREA             23258 non-null  str  
 12  STATUS           23258 non-null  str  
 13  MODALIDAD        23258 non-null  str  
 14  JORNADA          23258 non-null  str  
 15  PLAN             23258 non-null  str  
 16  DEPARTAMENTAL    23258 non-null  str  
dtypes: str(17)
memory usage: 3.0 MB


El dataset contiene 17 variables y 23,302 registros que proveen de información acerca de todos los centros educativos según el MINEDUC.

#### b. Tipo de dato de cada variable.



In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 23302 entries, 0 to 23301
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   CODIGO           23258 non-null  str  
 1   DISTRITO         22210 non-null  str  
 2   DEPARTAMENTO     23258 non-null  str  
 3   MUNICIPIO        23258 non-null  str  
 4   ESTABLECIMIENTO  23253 non-null  str  
 5   DIRECCION        22988 non-null  str  
 6   TELEFONO         20937 non-null  str  
 7   SUPERVISOR       22206 non-null  str  
 8   DIRECTOR         19915 non-null  str  
 9   NIVEL            23258 non-null  str  
 10  SECTOR           23258 non-null  str  
 11  AREA             23258 non-null  str  
 12  STATUS           23258 non-null  str  
 13  MODALIDAD        23258 non-null  str  
 14  JORNADA          23258 non-null  str  
 15  PLAN             23258 non-null  str  
 16  DEPARTAMENTAL    23258 non-null  str  
dtypes: str(17)
memory usage: 3.0 MB


Por la manera en la que se importaron los datos fue imposible conservar el formato o tipo original de cada variable, por lo que todas son reconocidas por Pandas como string. 

Sin embargo, es evidente que la variable, TELEFONO, es de tipo entero con 8 dígitos y CODIGO es un string que podría ser dividido en tres variables obtenidas al separar los números por guión.

#### c. Cantidad y porcentaje de valores faltantes por variable.



In [ ]:
valores_faltantes = df.isnull().sum()

# 2 promedio
porcentaje_faltantes = df.isnull().mean() * 100

# tabla
tabla_faltantes = pd.DataFrame({
    'Cantidad Faltantes': valores_faltantes,
    'Porcentaje (%)': porcentaje_faltantes
})
tabla_faltantes

#Ordenado por porcentaje de faltantes
tabla_faltantes = tabla_faltantes.sort_values(by="Porcentaje (%)", ascending=False)
tabla_faltantes


,Cantidad Faltantes,Porcentaje (%)
DIRECTOR,3387,14.535233
TELEFONO,2365,10.149343
SUPERVISOR,1096,4.703459
DISTRITO,1092,4.686293
DIRECCION,314,1.347524
ESTABLECIMIENTO,49,0.210282
CODIGO,44,0.188825
MUNICIPIO,44,0.188825
DEPARTAMENTO,44,0.188825
NIVEL,44,0.188825


Alrededor del 15% de los registros de DIRECTOR contienen datos faltantes, lo cual sugiere que existen falencias en el registro de datos debido a que cualquier instituto en Guatemala ha de tener un director. Según la Ley de Educación Nacional de Guatemala (Decreto 12-91), todo centro educativo, ya sea público, privado o por cooperativa, debe contar con un director. Esta figura es la máxima autoridad del establecimiento y la responsable legal, administrativa y pedagógica ante el Ministerio de Educación (MINEDUC).

Por otra parte, hacen falta un 10% de los teléfonos 

#### d. Cantidad de valores únicos.



In [8]:
df.nunique()
#23258*100/23302

CODIGO             23258
DISTRITO            1933
DEPARTAMENTO          22
MUNICIPIO            335
ESTABLECIMIENTO     8643
DIRECCION          12564
TELEFONO           11804
SUPERVISOR          1467
DIRECTOR           10581
NIVEL                  2
SECTOR                 4
AREA                   3
STATUS                 5
MODALIDAD              2
JORNADA                6
PLAN                  13
DEPARTAMENTAL         26
dtype: int64

La variable, CODIGO, está compuesta por un 99.81% de registros únicos; seguido por DIRECCION y TELEFONO como otras variables con mayor cantidad de valores únicos. La variable d DISTRITO contiene 1933. Se esperaba obtener 22 departamentos y municipios solamente hay 335 según el dataset. 

En total existen 13 planes educativos, 5 tipos de jornadas, 2 modalidades, hay 5 status distintos, 3 áreas, 4 sectores y solamente 2 niveles (básico y diversificado) por la restricción implementada al importar datos.

#### e. Cantidad de registros duplicados exactos.



In [9]:
resumen_duplicados = pd.DataFrame({
    'Métrica': [
        'Duplicados exactos (todas las ocurrencias, keep=False)',
        'Duplicados exactos (sin la primera ocurrencia)',
        'Filas completamente vacías (todas las columnas nulas)',
        'Duplicados exactos entre registros CON datos',
        'Códigos (CODIGO) repetidos entre registros no nulos',
    ],
    'Cantidad': [
        df.duplicated(keep=False).sum(),
        df.duplicated().sum(),
        df.isnull().all(axis=1).sum(),
        df[~df.isnull().all(axis=1)].duplicated().sum(),
        df[df['CODIGO'].notna()].duplicated(subset=['CODIGO']).sum(),
    ]
})
resumen_duplicados

,Métrica,Cantidad
0,"Duplicados exactos (todas las ocurrencias, kee...",44
1,Duplicados exactos (sin la primera ocurrencia),43
2,Filas completamente vacías (todas las columnas...,44
3,Duplicados exactos entre registros CON datos,0
4,Códigos (CODIGO) repetidos entre registros no ...,0


`duplicated()` marca 43 registros como duplicados exactos (44 si se cuentan todas las ocurrencias con `keep=False`). Sin embargo, al revisarlos se determinó que **la totalidad corresponde a filas completamente vacías** (sin ningún valor en sus 17 columnas): son las mismas ~44 filas responsables de los valores faltantes reportados en el inciso c, lo cual explica por qué casi todas las variables comparten un 0.19% de nulos.

Al excluir esas filas vacías, no queda ningún duplicado exacto entre establecimientos con información real, y la variable CODIGO —que actúa como identificador— no presenta ningún valor repetido entre los registros no nulos. Este hallazgo también implica que las filas vacías deberán eliminarse (o al menos documentarse) en el plan de limpieza, ya que están inflando artificialmente tanto el conteo de faltantes como el de duplicados.

#### f. Variables con valores fuera de dominio o inconsistentes.



In [10]:
from IPython.display import display

def tabla_conteo(serie, nombre_col):
    return serie.value_counts(dropna=False).rename_axis(nombre_col).reset_index(name='Cantidad')

print("AREA: valores fuera del dominio esperado {URBANA, RURAL}")
display(tabla_conteo(df['AREA'], 'AREA'))

print("NIVEL: se esperaba solo DIVERSIFICADO según el alcance del proyecto")
display(tabla_conteo(df['NIVEL'], 'NIVEL'))

print("STATUS: categorías y su frecuencia")
display(tabla_conteo(df['STATUS'], 'STATUS'))

telefonos = df['TELEFONO'].dropna().astype(str)
solo_digitos = telefonos.str.fullmatch(r"\d+")
telefono_valido = solo_digitos & (telefonos.str.len() == 8)
patron_codigo = df['CODIGO'].dropna().str.fullmatch(r"\d{2}-\d{2}-\d{4}-\d{2}")

print("TELEFONO y CODIGO: cumplimiento de dominio esperado")
resumen_dominio = pd.DataFrame({
    'Variable': ['TELEFONO', 'TELEFONO', 'CODIGO'],
    'Regla de dominio': [
        'Solo caracteres numéricos',
        'Exactamente 8 dígitos (formato guatemalteco)',
        'Patrón NN-NN-NNNN-NN',
    ],
    'Cumplen': [solo_digitos.sum(), telefono_valido.sum(), patron_codigo.sum()],
    'NO cumplen': [(~solo_digitos).sum(), len(telefonos) - telefono_valido.sum(), (~patron_codigo).sum()],
    'Total evaluado': [len(telefonos), len(telefonos), len(patron_codigo)],
})
display(resumen_dominio)

print("TELEFONO: distribución de longitud entre los valores puramente numéricos")
tabla_longitud = telefonos[solo_digitos].str.len().value_counts().rename_axis('N° de dígitos').reset_index(name='Cantidad').sort_values('N° de dígitos')
display(tabla_longitud)

AREA: valores fuera del dominio esperado {URBANA, RURAL}


,AREA,Cantidad
0,URBANA,14655
1,RURAL,8601
2,NaN,44
3,SIN ESPECIFICAR,2


NIVEL: se esperaba solo DIVERSIFICADO según el alcance del proyecto


,NIVEL,Cantidad
0,BASICO,13552
1,DIVERSIFICADO,9706
2,NaN,44


STATUS: categorías y su frecuencia


,STATUS,Cantidad
0,ABIERTA,15186
1,CERRADA TEMPORALMENTE,4871
2,CERRADA DEFINITIVAMENTE,3047
3,TEMPORAL NOMBRAMIENTO,110
4,NaN,44
5,TEMPORAL TITULOS,44


TELEFONO y CODIGO: cumplimiento de dominio esperado


,Variable,Regla de dominio,Cumplen,NO cumplen,Total evaluado
0,TELEFONO,Solo caracteres numéricos,20688,249,20937
1,TELEFONO,Exactamente 8 dígitos (formato guatemalteco),20616,321,20937
2,CODIGO,Patrón NN-NN-NNNN-NN,23258,0,23258


TELEFONO: distribución de longitud entre los valores puramente numéricos


,N° de dígitos,Cantidad
6,1,2
8,2,1
5,4,2
4,5,2
2,6,12
1,7,49
0,8,20616
3,9,2
7,10,1
9,16,1


AREA incluye la categoría **"SIN ESPECIFICAR"** (2 registros), fuera del dominio binario esperado {URBANA, RURAL}. NIVEL contiene **BASICO** (13,552 registros) además de DIVERSIFICADO (9,706), lo cual está fuera del alcance declarado por el proyecto (solo centros que llegan a diversificado); esto sugiere que la extracción de datos no filtró exclusivamente por ese nivel y deberá decidirse en el plan de limpieza si se filtra o se documenta como parte del universo de datos. STATUS presenta dos categorías poco frecuentes (TEMPORAL NOMBRAMIENTO y TEMPORAL TITULOS, con 110 y 44 registros) que son válidas pero merecen revisión por su baja representación.

La variable con más problemas de dominio es **TELEFONO**: 249 registros contienen caracteres no numéricos (guiones, comas o texto como "CEL."), y entre los valores puramente numéricos hay longitudes de 1, 2, 4, 5, 6, 7, 9, 10 y hasta 16 dígitos en lugar de los 8 dígitos esperados para un teléfono guatemalteco.

En contraste, DEPARTAMENTO (validado en el inciso d) y CODIGO sí cumplen su dominio esperado: los 22 valores de DEPARTAMENTO coinciden exactamente con los 22 departamentos oficiales de Guatemala, y el 100% de los códigos no nulos siguen el patrón NN-NN-NNNN-NN.

#### g. Variables con formatos inconsistentes.



In [11]:
from IPython.display import display

print("TELEFONO: ejemplos de formatos mixtos (guiones, comas, texto embebido)")
no_numericos = telefonos[~solo_digitos]
tabla_telefono_ejemplos = no_numericos.head(10).reset_index(drop=True).rename('Valor original').to_frame()
display(tabla_telefono_ejemplos)

est = df['ESTABLECIMIENTO'].dropna().astype(str)
print("ESTABLECIMIENTO: estilo de comillas usado para encerrar apodos")
tabla_comillas = pd.DataFrame({
    'Estilo de comillas': ['Dobles ("...")', "Simples ('...')"],
    'Cantidad': [est.str.contains(chr(34)).sum(), est.str.contains(chr(39)).sum()],
})
display(tabla_comillas)

dire = df['DIRECCION'].dropna().astype(str)
con_minusculas = dire.str.contains('[a-z]', regex=True)
print("DIRECCION: consistencia de mayúsculas")
tabla_direccion = pd.DataFrame({
    'Métrica': ['Con alguna letra minúscula', 'Total de direcciones no nulas'],
    'Cantidad': [con_minusculas.sum(), len(dire)],
})
display(tabla_direccion)

print("DIRECCION: ejemplos con minúsculas (abreviaturas de ordinal inconsistentes)")
tabla_direccion_ejemplos = dire[con_minusculas].head(5).reset_index(drop=True).rename('Valor original').to_frame()
display(tabla_direccion_ejemplos)

TELEFONO: ejemplos de formatos mixtos (guiones, comas, texto embebido)


,Valor original
0,66420121-66420111
1,66415990-56652813
2,"6372043, 6372123"
3,6400403 CEL. 4019799
4,24333306-24342926
5,24310907-24372550
6,"5987951, 3088873"
7,948849 592-44-33
8,24854408-24854398
9,"24431649, 24431650"


ESTABLECIMIENTO: estilo de comillas usado para encerrar apodos


,Estilo de comillas,Cantidad
0,"Dobles (""..."")",3696
1,Simples ('...'),1526


DIRECCION: consistencia de mayúsculas


,Métrica,Cantidad
0,Con alguna letra minúscula,19
1,Total de direcciones no nulas,22988


DIRECCION: ejemplos con minúsculas (abreviaturas de ordinal inconsistentes)


,Valor original
0,6a. CALLE 48-98 ZONA 7 COLONIA EL ROSARIO
1,"7A AVE. 13-30 COLONIA 1x. DE JULIO, ZONA 5"
2,"7A AVE. 13-30 COLONIA 1x. DE JULIO, ZONA 5"
3,4a CALLE 0-05 ZONA 1
4,"9a. AV. FINAL, ZONA 6, COL. SAN JOSE LOS PINOS..."


Además de los problemas de dominio, TELEFONO mezcla distintos formatos dentro de la misma columna: números separados por guion (p. ej. "66420121-66420111"), por coma (p. ej. "6372043, 6372123") o acompañados de texto (p. ej. "6400403 CEL. 4019799"), lo que indica que el campo a veces almacena más de un contacto o un número con anotación en lugar de un solo teléfono.

ESTABLECIMIENTO no tiene un estilo uniforme para encerrar apodos o nombres comerciales: 3,696 registros usan comillas dobles y 1,526 usan comillas simples. DIRECCION es casi enteramente uniforme en mayúsculas (solo 19 de 22,988 registros contienen alguna letra minúscula), pero esos mismos registros muestran una abreviatura inconsistente del ordinal de calle/avenida (p. ej. "6a. CALLE" en minúscula frente a la convención dominante "6A. CALLE"), evidenciando que la normalización de texto (inciso 5c del proyecto) tendrá que estandarizar mayúsculas y el uso de puntos en las abreviaturas.

#### h. Identificación de problemas potenciales de calidad de datos.



In [12]:
import unicodedata
from IPython.display import display

def normalizar(texto):
    return ''.join(c for c in unicodedata.normalize('NFD', str(texto)) if unicodedata.category(c) != 'Mn').upper().strip()

# DEPARTAMENTAL subdivide GUATEMALA en 4 zonas y QUICHE en 2; validamos que
# siempre correspondan al DEPARTAMENTO correcto (consistencia entre variables)
sub = df[['DEPARTAMENTO', 'DEPARTAMENTAL']].dropna().copy()
sub['dep_norm'] = sub['DEPARTAMENTO'].map(normalizar)
sub['depal_norm'] = sub['DEPARTAMENTAL'].map(normalizar)
sub['coincide'] = sub.apply(lambda r: r['depal_norm'].startswith(r['dep_norm']), axis=1)

print("Consistencia cruzada: DEPARTAMENTO vs. DEPARTAMENTAL")
tabla_consistencia = pd.DataFrame({
    'Métrica': ['Registros comparados', 'Contradicciones encontradas'],
    'Cantidad': [len(sub), (~sub['coincide']).sum()],
})
display(tabla_consistencia)

print("Síntesis de problemas potenciales de calidad de datos")
resumen_problemas = pd.DataFrame([
    {'Problema': 'Filas completamente vacías', 'Variable(s)': 'Todas', 'Cantidad': 44,
     'Detalle': 'Inflan los conteos de faltantes (c) y de duplicados exactos (e)'},
    {'Problema': 'Tipo de dato incorrecto', 'Variable(s)': 'TELEFONO, CODIGO', 'Cantidad': 23302,
     'Detalle': 'Todas las columnas se leyeron como texto (b)'},
    {'Problema': 'Formato/longitud inválida', 'Variable(s)': 'TELEFONO', 'Cantidad': 249,
     'Detalle': 'Guiones, comas, texto embebido; longitudes de 1 a 28 en vez de 8 dígitos'},
    {'Problema': 'Fuera del alcance declarado', 'Variable(s)': 'NIVEL', 'Cantidad': 13552,
     'Detalle': 'Incluye BASICO; el proyecto pide solo nivel DIVERSIFICADO'},
    {'Problema': 'Categoría residual fuera de dominio', 'Variable(s)': 'AREA', 'Cantidad': 2,
     'Detalle': '"SIN ESPECIFICAR" no pertenece a {URBANA, RURAL}'},
    {'Problema': 'Formato de texto no uniforme', 'Variable(s)': 'ESTABLECIMIENTO, DIRECCION', 'Cantidad': 5241,
     'Detalle': 'Comillas simples/dobles mezcladas; mayúsc./minúsc. en abreviaturas de calle'},
    {'Problema': 'Consistencia entre variables', 'Variable(s)': 'DEPARTAMENTO / DEPARTAMENTAL', 'Cantidad': 0,
     'Detalle': 'Sin contradicciones — hallazgo positivo, no requiere corrección'},
])
display(resumen_problemas)

Consistencia cruzada: DEPARTAMENTO vs. DEPARTAMENTAL


,Métrica,Cantidad
0,Registros comparados,23258
1,Contradicciones encontradas,0


Síntesis de problemas potenciales de calidad de datos


,Problema,Variable(s),Cantidad,Detalle
0,Filas completamente vacías,Todas,44,Inflan los conteos de faltantes (c) y de dupli...
1,Tipo de dato incorrecto,"TELEFONO, CODIGO",23302,Todas las columnas se leyeron como texto (b)
2,Formato/longitud inválida,TELEFONO,249,"Guiones, comas, texto embebido; longitudes de ..."
3,Fuera del alcance declarado,NIVEL,13552,Incluye BASICO; el proyecto pide solo nivel DI...
4,Categoría residual fuera de dominio,AREA,2,"""SIN ESPECIFICAR"" no pertenece a {URBANA, RURAL}"
5,Formato de texto no uniforme,"ESTABLECIMIENTO, DIRECCION",5241,Comillas simples/dobles mezcladas; mayúsc./min...
6,Consistencia entre variables,DEPARTAMENTO / DEPARTAMENTAL,0,"Sin contradicciones — hallazgo positivo, no re..."


Al cruzar DEPARTAMENTO con DEPARTAMENTAL no se encontró ninguna contradicción: las 26 categorías de DEPARTAMENTAL siempre corresponden al departamento correcto en DEPARTAMENTO. Es una validación cruzada exitosa entre dos variables relacionadas, y descarta que la granularidad extra de DEPARTAMENTAL (visible en el inciso d como "26 valores únicos" frente a los 22 departamentos) sea un error de inconsistencia; es, en realidad, una subdivisión administrativa válida (Guatemala en 4 zonas, Quiché en 2).

En síntesis, los principales problemas de calidad identificados en este diagnóstico son:

1. **Filas completamente vacías** (44 registros, ~0.19%) que inflan artificialmente los conteos de valores faltantes (inciso c) y de duplicados exactos (inciso e).
2. **Tipos de dato incorrectos**: todas las columnas fueron leídas como texto (inciso b); TELEFONO debería ser numérico y CODIGO podría descomponerse en variables categóricas (departamento-distrito-consecutivo-nivel).
3. **TELEFONO es el campo más problemático**: mezcla números concatenados, separadores inconsistentes (guion, coma, espacio), texto embebido ("CEL.") y longitudes que van de 1 a 28 caracteres en lugar de los 8 dígitos esperados.
4. **NIVEL incluye BASICO** pese a que el alcance del proyecto solo pide establecimientos de nivel diversificado — deberá decidirse si se filtra o se documenta como parte del universo de datos.
5. **Categoría residual "SIN ESPECIFICAR"** en AREA, fuera del dominio binario esperado (URBANA/RURAL).
6. **Formato de texto no uniforme** en ESTABLECIMIENTO (comillas simples vs. dobles) y en DIRECCION (abreviaturas de ordinal en mayúscula/minúscula), aunque en general el dataset ya está mayormente normalizado a mayúsculas.
7. **Ausencia de contradicciones** entre DEPARTAMENTO y DEPARTAMENTAL, y entre los 22 valores de DEPARTAMENTO y la división política oficial del país — puntos positivos que no requieren corrección.

Estos hallazgos alimentarán directamente el plan de limpieza (actividad 4), donde deberá definirse la regla de corrección para cada uno de estos problemas.